Update release notes for Foreman 3.3.0-rc1 (#1983)
theforeman/theforeman.org: _includes/manuals/3.3/1.2_release_notes.md

* Mozilla Firefox - latest version
* Mozilla Firefox Extended Support Release (ESR) - latest version

### Release notes for 3.3.0
#### Foreman

> * Support EC (and other non-RSA) keys in foreman core ([#34843](https://projects.theforeman.org/issues/34843))

Reviewer: ```suggestion
* Support EC (and other non-RSA) keys in Foreman core ([#34843](https://projects.theforeman.org/issues/34843))
```

> * Remove Share your feedback link in new host page

Reviewer: ```suggestion
* Remove the **Share your feedback** link from the new host details page
```

 ([#34841](https://projects.theforeman.org/issues/34841))

> * Add host page reports tab  ([#34782](https://projects.theforeman.org/issues/34782))

Reviewer: ```suggestion
* Add a reports tab to the hosts details page ([#34782](https://projects.theforeman.org/issues/34782))
```

* Add developer documentation for pagelets ([#34544](https://projects.theforeman.org/issues/34544))

> * Host list page should show the host comment ([#34516](https://projects.theforeman.org/issues/34516))

Reviewer: Should it? Does it? I don't understand.

Author: how about: Include the host comments to a host list page

> * Host list page should show the owner of the host ([#34515](https://projects.theforeman.org/issues/34515))

Reviewer: Same here. 
Do or do not, there is no should ;-)

Author: > Same here. Do or do not, there is no should ;-)

Include the host's owner to a host list page
?

Reviewer: That will work :)

#### Foreman - API

> * Taxonomies api dont accept per_page all ([#34748](https://projects.theforeman.org/issues/34748))

Reviewer: ```suggestion
* Taxonomies api does not accept per_page all ([#34748](https://projects.theforeman.org/issues/34748))
```

* Refactor handling of External IPAM response ([#34714](https://projects.theforeman.org/issues/34714))
* Turn off caching for apipie in production ([#34643](https://projects.theforeman.org/issues/34643))
* Improve GraphQL association resolving ([#34557](https://projects.theforeman.org/issues/34557))
* Add import_ipv4_subnets to API ([#34251](https://projects.theforeman.org/issues/34251))

#### Foreman - Audit Log

> * Add rake that print out documentation of Auditable attributes ([#34690](https://projects.theforeman.org/issues/34690))

Reviewer: ```suggestion
* Add rake command that prints out documentation for Auditable attributes ([#34690](https://projects.theforeman.org/issues/34690))
```

#### Foreman - Dashboard
* Setting all_out_of_sync_disabled has no definition warning ([#34240](https://projects.theforeman.org/issues/34240))

#### Foreman - Database
* Undefined method 'format_errors' when db:seed fail ([#34513](https://projects.theforeman.org/issues/34513))
* Ignore dynflow tables for schema dump ([#33660](https://projects.theforeman.org/issues/33660))

#### Foreman - Facts
* Shorten DNS timeout for primary NIC detection ([#34462](https://projects.theforeman.org/issues/34462))
* Normalize fact parsers to use CentOS instead of centos ([#34450](https://projects.theforeman.org/issues/34450))

#### Foreman - Host creation
* Installation Media does not find Ubuntu autoinstall kernel files ([#34565](https://projects.theforeman.org/issues/34565))
* Ubuntu Autoinstall support ([#32632](https://projects.theforeman.org/issues/32632))
* Reimplement !unattended mode support as mainline feature ([#10413](https://projects.theforeman.org/issues/10413))

#### Foreman - Host registration
* Registration migration with nil template ([#34661](https://projects.theforeman.org/issues/34661))

#### Foreman - Inventory

> * Submit button on Edit page of a host will revert back to a invalid page on Satellite ([#34830](https://projects.theforeman.org/issues/34830))

Reviewer: ```suggestion
* Submit button on Edit page of a host will revert back to a invalid page on Foreman ([#34830](https://projects.theforeman.org/issues/34830))
```

* Submit button on host form redirects to invalid page ([#34572](https://projects.theforeman.org/issues/34572))
* Expose reported data in the API ([#34391](https://projects.theforeman.org/issues/34391))

#### Foreman - JavaScript stack

> * show error when using node 16+ ([#34857](https://projects.theforeman.org/issues/34857))

Reviewer: ```suggestion
* Show error when using node 16+ ([#34857](https://projects.theforeman.org/issues/34857))
```

* Fix slots in system properties card ([#34815](https://projects.theforeman.org/issues/34815))
* New host detail page - sentence case fixes ([#34798](https://projects.theforeman.org/issues/34798))
* new host detail page - Fix host status card alignment ([#34755](https://projects.theforeman.org/issues/34755))
* Adding host parameters clears the values of the prior line ([#34747](https://projects.theforeman.org/issues/34747))
* Better handling when no key is passed to API middleware ([#34745](https://projects.theforeman.org/issues/34745))

> * Remove pf3 OverlayTrigger ([#34713](https://projects.theforeman.org/issues/34713))

Reviewer: ```suggestion
* Remove PatternFly 3 OverlayTrigger ([#34713](https://projects.theforeman.org/issues/34713))
```

* Use hostFriendlyId and hostName props for host ActionBar component ([#34673](https://projects.theforeman.org/issues/34673))
* Provide helpers for dealing with global IDs returned by graphql queries ([#34614](https://projects.theforeman.org/issues/34614))
* Remove pagination deprecations ([#34580](https://projects.theforeman.org/issues/34580))
* Add ouiaId on components for easier testing ([#34542](https://projects.theforeman.org/issues/34542))
* Update to the latest foreman-js ([#34481](https://projects.theforeman.org/issues/34481))

> * Create pf4 breadcrumb switcher ([#34290](https://projects.theforeman.org/issues/34290))

Reviewer: ```suggestion
* Create PatternFly 4 breadcrumb switcher ([#34290](https://projects.theforeman.org/issues/34290))
```

* useAPI infinite loop ([#33236](https://projects.theforeman.org/issues/33236))

#### Foreman - Organizations and Locations
* Organization context fails to change in web UI ([#34416](https://projects.theforeman.org/issues/34416))

#### Foreman - Plugin integration
* Facets do not allow emptying their relations through mass assignment ([#34375](https://projects.theforeman.org/issues/34375))

#### Foreman - Power management
* Disable EFI local chainloading by default ([#34532](https://projects.theforeman.org/issues/34532))

#### Foreman - Rails
* Upgrade Rails to 6.0.4.7 ([#34649](https://projects.theforeman.org/issues/34649))
* Move initialization from application.rb into initializer ([#34646](https://projects.theforeman.org/issues/34646))

> * order description syntax does not read right with Maruku ([#34634](https://projects.theforeman.org/issues/34634))

Reviewer: ```suggestion
* Order description syntax does not read right with Maruku ([#34634](https://projects.theforeman.org/issues/34634))
```

* SettingPresenter - pass kwarks in Ruby 3 compatible format ([#34570](https://projects.theforeman.org/issues/34570))
* Postpone LookupValue match validations ([#34569](https://projects.theforeman.org/issues/34569))
* Switch Rails version to 6.1 ([#34526](https://projects.theforeman.org/issues/34526))
* Add support for Rails 6.1 ([#34500](https://projects.theforeman.org/issues/34500))

> * remove usage of force_ssl in controller ([#30122](https://projects.theforeman.org/issues/30122))

Reviewer: ```suggestion
* Remove usage of force_ssl in controller ([#30122](https://projects.theforeman.org/issues/30122))
```

* Report template request: list enabled repositories and CV package counts ([#34712](https://projects.theforeman.org/issues/34712))
* "Subscription - Entitlement Report" does not show correct number of subscriptions attached/consumed ([#34610](https://projects.theforeman.org/issues/34610))

#### Foreman - Security
* Settings defined by DSL are not properly encrypted ([#34573](https://projects.theforeman.org/issues/34573))

#### Foreman - Settings
* Settings - Issues with search ([#34866](https://projects.theforeman.org/issues/34866))
* "Sync Connect Timeout" settings takes invalid value and shows update successful but doesn't reflect the change for invalid values ([#34838](https://projects.theforeman.org/issues/34838))
* Deprecate other then DSL setting definitions ([#34603](https://projects.theforeman.org/issues/34603))
* GraphQL is using Setting model directly ([#34328](https://projects.theforeman.org/issues/34328))
* Initialize Setting for tests without fixtures ([#33782](https://projects.theforeman.org/issues/33782))
* Move setting value collections to SettingPresenter ([#30861](https://projects.theforeman.org/issues/30861))

#### Foreman - Statistics

> * Host - Last Checkin report template is not showing any other content host apart from Red Hat Satellite itself.

Reviewer: ```suggestion
* Host - Last Checkin report template is not showing any other content host apart from Foreman itself.
```

* Make operatingsystem.bootfile useable in Jail ([#34689](https://projects.theforeman.org/issues/34689))

#### Foreman - Templates
* Use foreman request address in windows iPXE  ([#34710](https://projects.theforeman.org/issues/34710))
* Harmonize preseed templates ([#34658](https://projects.theforeman.org/issues/34658))
* Ensure that the insights snippet is being called by honoring the value of host_registration_insights parameter ([#34525](https://projects.theforeman.org/issues/34525))
* AutoYaST PXE templates fail to render if http-proxy parameter is set without http-proxy-port ([#34489](https://projects.theforeman.org/issues/34489))
* Allow puppet setup to be skipped even if you set a puppet master ([#34388](https://projects.theforeman.org/issues/34388))
* AutoYaST SLES template invalid for SLES 15 SP3 ([#34311](https://projects.theforeman.org/issues/34311))
* Support unattented partitionning using crypto under Debian ([#34307](https://projects.theforeman.org/issues/34307))
* Add driverdisk support to kickstart templates ([#33938](https://projects.theforeman.org/issues/33938))
* Cannot change "local boot ipxe template" ([#33937](https://projects.theforeman.org/issues/33937))

> * templates - vgname is hardcoded to vg_sda ([#33930](https://projects.theforeman.org/issues/33930))

Reviewer: ```suggestion
* Templates - vgname is hardcoded to vg_sda ([#33930](https://projects.theforeman.org/issues/33930))
```

* Add DSL autocompletion in templates ([#32035](https://projects.theforeman.org/issues/32035))
* Request to add UEFI Grub2 for SLES/SUSE ([#20265](https://projects.theforeman.org/issues/20265))

#### Foreman - Tests
* Sanitize fqdn in tests ([#34339](https://projects.theforeman.org/issues/34339))
* Improve a clarity of Api::HostsController test ([#34326](https://projects.theforeman.org/issues/34326))

#### Foreman - Unattended installations
* Default boot template names ([#34596](https://projects.theforeman.org/issues/34596))
* Trigger ansible provisioning callback for Preseed based installs ([#34558](https://projects.theforeman.org/issues/34558))

> * Use systemd based ansible callback on Ubuntu &gt;= 15 and Debian &gt;= 8 ([#34553](https://projects.theforeman.org/issues/34553))

Reviewer: ```suggestion
* Use systemd based Ansible callback on Ubuntu &gt;= 15 and Debian &gt;= 8 ([#34553](https://projects.theforeman.org/issues/34553))
```

#### Foreman - Users, Roles and Permissions

> * Infer of permission name for isolated namespace controllers doesnt work ([#34506](https://projects.theforeman.org/issues/34506))

Reviewer: I don't think this makes sense but don't know how to fix it.

Author: As per the pull request of redmine I can formulate something: The authorized_for doesn't work for isolated namespace engines

#### Foreman - Web Interface
* Change host status icon to green ([#34831](https://projects.theforeman.org/issues/34831))
* Tabbable latest version 5.3.1 is not compatible with jest dom/ JSDom without changes to PF4. ([#34802](https://projects.theforeman.org/issues/34802))
* Breadcrumb switcher use V for selected ([#34662](https://projects.theforeman.org/issues/34662))
* Cancel host form redirects to legacy host UI ([#34579](https://projects.theforeman.org/issues/34579))
* Add pagelet mountpoints for hosts list table ([#34543](https://projects.theforeman.org/issues/34543))
* Add breadcrumbs switcher to new host page ([#34511](https://projects.theforeman.org/issues/34511))
* Breadcrumb switcher doesn't work with katello content ([#34495](https://projects.theforeman.org/issues/34495))

> * close bookmarks dropdown after selecting a bookmark ([#34228](https://projects.theforeman.org/issues/34228))

Reviewer: ```suggestion
* Close bookmarks dropdown after selecting a bookmark ([#34228](https://projects.theforeman.org/issues/34228))
```

#### Hammer
* Fix hammer shell with pipe as stdin ([PR #360](https://github.com/theforeman/hammer-cli/pull/360)), [#34724](http://projects.theforeman.org/issues/34724)
* Add kerberos negotiate auth support ([PR #555](https://github.com/theforeman/hammer-cli-foreman/pull/555)), [#8923](http://projects.theforeman.org/issues/8923)
* Pin mocha gem to < 1.14.0
* Force api docs checksum check, [#28283](http://projects.theforeman.org/issues/28283)
* Add template report-remplate and partition-table export command ([PR #595](https://github.com/theforeman/hammer-cli-foreman/pull/595)), [#34503](http://projects.theforeman.org/issues/34503)
* Add template import and partition-table import commands ([PR #596](https://github.com/theforeman/hammer-cli-foreman/pull/596)), [#22692](http://projects.theforeman.org/issues/22692)
* Add resource information to download command ([PR #598](https://github.com/theforeman/hammer-cli-foreman/pull/598)), [#34621](http://projects.theforeman.org/issues/34621)
* Add command to import ipv4 subnet from smart proxy ([PR #593](https://github.com/theforeman/hammer-cli-foreman/pull/593)), [#33255](http://projects.theforeman.org/issues/33255)

#### Installer

> * installer spams with katello-certs-check output when using custom certs

Reviewer: ```suggestion
* Installer spams with katello-certs-check output when using custom certs
```

 ([#34888](https://projects.theforeman.org/issues/34888))

> * display the mismatched FQDN additionally rather than just showing the commands to verify the output ([#34883](https://projects.theforeman.org/issues/34883))

Reviewer: ```suggestion
* Display the mismatched FQDN additionally rather than just showing the commands to verify the output ([#34883](https://projects.theforeman.org/issues/34883))
```

* Pulp: Add options to change the import and export path in  /etc/pulp/settings.py ([#34882](https://projects.theforeman.org/issues/34882))
* Resetting nssdb certificate does not update private key and breaks Qpid ([#34860](https://projects.theforeman.org/issues/34860))

> * installer does not restart foreman.service when changing puma configuration ([#34824](https://projects.theforeman.org/issues/34824))

Reviewer: ```suggestion
* Installer does not restart foreman.service when changing Puma configuration ([#34824](https://projects.theforeman.org/issues/34824))
```

* Puppet Agent enabled in Katello installations, even if it should not ([#34819](https://projects.theforeman.org/issues/34819))
* Rename foreman_proxy::plugin::remote_execution::ssh to foreman_proxy::plugin::remote_execution::script ([#34758](https://projects.theforeman.org/issues/34758))
* After upgrading to Katello 4.0+ ping check fails with "Some components are failing: katello_agent"
 ([#34708](https://projects.theforeman.org/issues/34708))
* --reset-data does not remove /var/lib/candlepin/.puppet-candlepin-rpm-version ([#34686](https://projects.theforeman.org/issues/34686))
* Drop apipie cache generation and indexing ([#34640](https://projects.theforeman.org/issues/34640))

> * detect plugin installation and trigger puma restart ([#34602](https://projects.theforeman.org/issues/34602))

Reviewer: ```suggestion
* Detect plugin installation and trigger Puma restart ([#34602](https://projects.theforeman.org/issues/34602))
```

* Keytool does not work on FIPS enabled EL 8 ([#34598](https://projects.theforeman.org/issues/34598))
* Add hammer-cli-foreman-host-reports to the installer ([#34505](https://projects.theforeman.org/issues/34505))
* Add support for REX pull transport ([#34239](https://projects.theforeman.org/issues/34239))
* Warning: postgresql.service changed on disk. Run 'systemctl daemon-reload' to reload units. ([#32323](https://projects.theforeman.org/issues/32323))

> * make it possible to install Foreman's httpd with mpm_event ([#20889](https://projects.theforeman.org/issues/20889))

Reviewer: ```suggestion
* Make it possible to install Foreman's httpd with mpm_event ([#20889](https://projects.theforeman.org/issues/20889))
```

#### Installer - Foreman modules
* OS upgrade keeps original TFTP setup preventing machines to boot from the network ([#34774](https://projects.theforeman.org/issues/34774))

#### Installer - foreman-installer script
* Run Apache httpd without default modules from puppetlabs-apache ([#34590](https://projects.theforeman.org/issues/34590))

#### Packaging
* Require psql for foreman-maintain ([#34855](https://projects.theforeman.org/issues/34855))
* Update foreman-protector yum and dnf source file paths ([#34801](https://projects.theforeman.org/issues/34801))
* Drop apipie cache generation from RPM builds ([#34641](https://projects.theforeman.org/issues/34641))

#### Packaging - Debian/Ubuntu

> * Should update oauth Gem installed by puppet-agent-oauth under Debian to a newer version (0.5.1 produces lots of warnings about URI.escape) ([#34681](https://projects.theforeman.org/issues/34681))

Reviewer: ```suggestion
* Updates oauth Gem installed by puppet-agent-oauth under Debian to a newer version (0.5.1 produces lots of warnings about URI.escape) ([#34681](https://projects.theforeman.org/issues/34681))
```

#### Packaging - RPMs
* foreman.rpm pulls in nodejs ([#34507](https://projects.theforeman.org/issues/34507))

> * instaler uses hostname, but that package can be absent ([#34468](https://projects.theforeman.org/issues/34468))

Reviewer: ```suggestion
* Installer uses hostname, but that package can be absent ([#34468](https://projects.theforeman.org/issues/34468))
```
